mode.go

```diff
@@ -4,32 +4,39 @@ import (
 	"github.com/jroimartin/gocui"
 )
 
+type UIMode int
+
 // List of all modes
 const (
-	modeInsert = iota
-	modeOverwrite
-	modeEscape
-	modeConnect
+	modeInsert    UIMode = iota
+	modeOverwrite UIMode = iota
+	modeEscape    UIMode = iota
+	modeConnect   UIMode = iota
+	modeSetPing   UIMode = iota
 )
 
-var modeChars = []struct {
-	c  rune
-	bg gocui.Attribute
-}{
-	{' ', gocui.ColorGreen},
-	{'R', gocui.ColorGreen},
-	{' ', gocui.ColorRed},
-	{'c', gocui.ColorRed},
+type ModeStyle struct {
+	Char    rune
+	BgColor gocui.Attribute
+}
+
+var modeChars = map[UIMode]ModeStyle{
+	modeInsert:    ModeStyle{' ', gocui.ColorGreen},
+	modeOverwrite: ModeStyle{'R', gocui.ColorGreen},
+	modeEscape:    ModeStyle{' ', gocui.ColorRed},
+	modeConnect:   ModeStyle{'c', gocui.ColorRed},
+	modeSetPing:   ModeStyle{'p', gocui.ColorRed},
 }
```

Comment:
maps are slow here (the constants are sequential, it just adds a whole lot of unnecessary lookup and runtime here.) Change list of constants above to add at the end `modeMax`, and then change this variable to this? (placing the constants to specify the index in the array is unnecessary, but improves readability and ensures the reader the code is correct

```suggestion
var modeChars = [modeMax]ModeStyle{
	modeInsert:    ModeStyle{' ', gocui.ColorGreen},
	modeOverwrite: ModeStyle{'R', gocui.ColorGreen},
	modeEscape:    ModeStyle{' ', gocui.ColorRed},
	modeConnect:   ModeStyle{'c', gocui.ColorRed},
	modeSetPing:   ModeStyle{'p', gocui.ColorRed},
}
```